presto-main/src/test/java/io/prestosql/operator/scalar/TestDateTimeFunctions.java

```diff
@@ -340,7 +340,9 @@ public void testPartFunctions()
         assertFunction("minute(" + TIMESTAMP_LITERAL + ")", BIGINT, (long) TIMESTAMP.getMinuteOfHour());
         assertFunction("hour(" + TIMESTAMP_LITERAL + ")", BIGINT, (long) TIMESTAMP.getHourOfDay());
         assertFunction("day_of_week(" + TIMESTAMP_LITERAL + ")", BIGINT, (long) TIMESTAMP.dayOfWeek().get());
+        assertFunction("day_of_week(" + TIMESTAMP_LITERAL + ", 5)", BIGINT, (long) TIMESTAMP.dayOfWeek().get() + 3);
```

Comment:
I think we'd want to avoid hardcoding the assertion for the given TIMESTAMP value. With the suggested implementation, we can do that by writing `floorMod(TIMESTAMP.dayOfWeek().get() - 5, DAYS_IN_WEEK) + 1`

Reply:
Updated.